tests/telemetry/test_writer.py

```diff
@@ -380,19 +383,20 @@ def test_update_dependencies_event_when_disabled(telemetry_writer, test_agent_se
             assert event["request_type"] != "app-dependencies-loaded"
 
 
-@pytest.mark.skip(reason="FIXME: This test does not generate a dependencies event")
 def test_update_dependencies_event_not_stdlib(telemetry_writer, test_agent_session, mock_time):
     # Fetch modules to reset the state of seen modules
-    modules.get_newly_imported_modules()
+    modules.uninstall_import_hook()
+    modules.install_import_hook()
+    telemetry_writer.reset_queues()
+    assert telemetry_writer._imported_dependencies == {}
+    del sys.modules["httpretty"]
+    import httpretty  # noqa F401
```

Comment:
Maybe this should be tested in a subprocess. It feels like the kind of thing that could introduce weirdness with other tests that end up running in the same process.